### PR TITLE
Actually run shellcheck on CI

### PR DESCRIPTION
**Pull request by nabijaczleweli**
#23374

### CODE REVIEW

**Review by mrc0mmand — CHANGES_REQUESTED** [review]
I don't think this is correct. Answering the comment from https://github.com/systemd/systemd/issues/23374#issuecomment-1133902874 there are two factors that might be culprits here:

1) We do only differential shellcheck in each PR (`VALIDATE_ALL_CODEBASE=true`), i.e. we check only files changed by that PR

2) The "miss" in https://github.com/systemd/systemd/issues/23374#issuecomment-1133902874 was most ilkely caused by the  `FILTER_REGEX_EXCLUDE: .*/(man/.*|.*\.(in|SKELETON)|tools/coverity\.sh)$` regex, which ignores the template files (`*.in`) by default, since they are not usually compatible with shellcheck before they're processed by a templating engine

**Comment by nabijaczleweli** [comment]
In workable state now, IMO

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -37,14 +37,16 @@ jobs:
...
-          #        ubuntu-22.04 as a valid runner specification. This should
-          #        be fixed once https://github.com/github/super-linter/pull/2897
-          #        is merged, which includes
-          #        https://github.com/rhysd/actionlint/commit/09a60b0f5761c2d066def294e541c25a89756eea
+          #    the current version of super-linter doesn't recognize
+          #    ubuntu-22.04 as a valid runner specification. This should
+          #    be fixed once https://github.com/github/super-linter/pull/2897
+          #    is merged, which includes
+          #    https://github.com/rhysd/actionlint/commit/
+          #            09a60b0f5761c2d066def294e541c25a89756eea
```
Comment (keszybz) [inline .github/workflows/linter.yml]: Please drop this chunk.

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -37,14 +37,16 @@ jobs:
           #   - man/.* - all snippets in man pages (false positives due to
           #              missing shebangs)
           #   - .*\.(in|SKELETON) - all template/skeleton files
+          #                         except kernel-install
           #   - tools/coverity\.sh - external file (with some modifications)
-          FILTER_REGEX_EXCLUDE: .*/(man/.*|.*\.(in|SKELETON)|tools/coverity\.sh)$
+          FILTER_REGEX_EXCLUDE: .*/(man/.*|([^k]|k(k|ek)*([^ek]|e[^kr]))*(k(k|ek)*e?)?\.(in|SKELETON)|tools/coverity\.sh)$
```
Comment (keszybz) [inline .github/workflows/linter.yml:42]: This is scary ;) Can we maybe use a negative match here:
`…|(?<!kernel-install)\.(in|SKELETON)|…` ?
Don't know if this is supported by this engine.

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -37,14 +37,16 @@ jobs:
...
           VALIDATE_ALL_CODEBASE: false
-          VALIDATE_BASH: true
```
Comment (mrc0mmand) [inline .github/workflows/linter.yml:43]: Why this change? The validators default to `false` when at least one of them is set to `true`, so the original `VALIDATE_BASH: true` should work just fine.

File: .github/workflows/linter.yml [hunk 4]
```diff
@@ -37,14 +37,16 @@ jobs:
...
+          VALIDATE_SHELL_SHFMT: false
           # FIXME: temporarily disable validation of GH Action files, since
-          #        the current version of super-linter doesn't recognize
-          #        ubuntu-22.04 as a valid runner specification. This should
-          #        be fixed once https://github.com/github/super-linter/pull/2897
-          #        is merged, which includes
-          #        https://github.com/rhysd/actionlint/commit/09a60b0f5761c2d066def294e541c25a89756eea
```
Comment (mrc0mmand) [inline .github/workflows/linter.yml]: Unnecessary ident change?

File: .github/workflows/linter.yml [hunk 5]
```diff
@@ -37,14 +37,16 @@ jobs:
           #   - man/.* - all snippets in man pages (false positives due to
           #              missing shebangs)
           #   - .*\.(in|SKELETON) - all template/skeleton files
+          #                         except kernel-install
           #   - tools/coverity\.sh - external file (with some modifications)
-          FILTER_REGEX_EXCLUDE: .*/(man/.*|.*\.(in|SKELETON)|tools/coverity\.sh)$
```
Comment (mrc0mmand) [inline .github/workflows/linter.yml:41]: Oh well, this is unfortunate, but it is what it is, thanks for figuring this out.

**Review by mrc0mmand — COMMENTED** [review]
A couple of nits, otherwise LGTM, thanks!

**Comment by mrc0mmand** [comment]
(To test this out you'll have to explicitly "touch" the kernel-install script, since by reverting the fixup commit this PR effectively doesn't touch kernel-install at all.)

File: .github/workflows/linter.yml (hunk 4 above)
Comment (nabijaczleweli) [reply .github/workflows/linter.yml]: Right, the defaults table misled me, and the BASH=true -> SHFMT=false conversion made the YAML validator run, which exploded on >80-character lines; fixed

**Comment by nabijaczleweli** [comment]
Blegh, yeah; validated, fixed, should be g2g

**Review by mrc0mmand — APPROVED** [review]
Awesome, LGTM, thanks!

File: .github/workflows/linter.yml (hunk 2 above)
Comment (mrc0mmand) [reply .github/workflows/linter.yml:42]: Unfortunately, no, this uses Bash's POSIX regex engine (`regex(3)`), so no fancy lookaheads/lookbehinds here.

Comment (nabijaczleweli) [reply .github/workflows/linter.yml:42]: That's what I tried (and helpfully the documentation doesn't say; well it says it's RE, which it isn't), but it uses bash `[[ =~ ]]`, which is ERE, to match these, and EREs don't have lookaheads

**Pull request merged**